inc/GenInfo.hh

```diff
@@ -16,12 +16,13 @@ namespace mu2e
   struct GenInfo {
     Int_t _pdg, _gen; // true PDG code, generator code
     Float_t _time;  // time of this step
-    XYZVec _mom;   // momentum at the start of this step
+    XYZVec _dir;   // direction at the start of this step
```

Comment:
making dir an XYZVec is in principle correct, assuming we can access the functions (phi(), theta(), ...) as well as the data members in root.  Have you checked if those are available, or can you provide the libraries to load so that they work?  Do you know if that will work with uproot?  If not, I suggest to replace this with cos(theta) and phi, as those are the most relevant physical variables.

Reply:
As far as I know there is now to access the methods from uproot (see https://github.com/scikit-hep/uproot3/issues/418). If there is a file available produced with this version I can look at it and verify what works and what doesn't.

Reply:
It looks like we only write out the three floats of XYZVec and not the XYZVec itself so there's no access to the methods. I will change it to cos(theta) and phi.